cmd/nerdctl/run.go

```diff
@@ -737,6 +739,42 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 	return container, nil, nil
 }
 
+func processPullCommandFlagsInRun(cmd *cobra.Command) (types.PullCommandOptions, error) {
+	verifier, err := cmd.Flags().GetString("verify")
+	if err != nil {
+		return types.PullCommandOptions{}, err
+	}
+	cosignKey, err := cmd.Flags().GetString("cosign-key")
+	if err != nil {
+		return types.PullCommandOptions{}, err
+	}
+	ipfsAddressStr, err := cmd.Flags().GetString("ipfs-address")
+	if err != nil {
+		return types.PullCommandOptions{}, err
+	}
+	return types.PullCommandOptions{
+		Verify:      verifier,
+		CosignKey:   cosignKey,
+		IPFSAddress: ipfsAddressStr,
+	}, nil
+}
+
+func ensureImage(ctx context.Context, cmd *cobra.Command, client *containerd.Client, globalOptions types.GlobalCommandOptions, rawRef string, ocispecPlatforms []v1.Platform, pull string, unpack *bool, quiet bool) (*imgutil.EnsuredImage, error) {
+	options, err := processPullCommandFlagsInRun(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	options.GOptions = globalOptions
+
+	ensured, err := image.EnsureImage(ctx, client, rawRef, cmd.OutOrStdout(), cmd.ErrOrStderr(), options, ocispecPlatforms, pull, unpack, quiet)
+	if err != nil {
+		return nil, err
+	}
+
+	return ensured, nil
+}
```

Comment:
you can remove the helper func `ensureImage`, and call `image.EnsureImage` directly in `generateRootfsOpts`. E.g,: change https://github.com/containerd/nerdctl/blob/2fa1777ceef949786ff1f730c0ad8ef0c7670dc9/cmd/nerdctl/run.go#L763

to

```golang
	options, err := processPullCommandFlagsInRun(cmd)
	if err != nil {
		return nil, err
	}
	options.GOptions = globalOptions
    ensured, err := image.EnsureImage(ctx, client, rawRef, cmd.OutOrStdout(), cmd.ErrOrStderr(), options, ocispecPlatforms, pull, unpack, quiet)
```

Reply:
Make sense. It looks a little more cohesive.

Reply:
Done.